apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/logs/_components/DocumentLogs/ExportLogsModal/Content.tsx

```diff
@@ -0,0 +1,47 @@
+import { useEffect } from 'react'
+
+import { CsvData } from '@latitude-data/core/browser'
+import { TableSkeleton, Text } from '@latitude-data/web-ui'
+import { CsvPreviewTable } from '$/app/(private)/datasets/generate/CsvPreviewTable'
+import useFetcher from '$/hooks/useFetcher'
+import { ROUTES } from '$/services/routes'
+import useSWR from 'swr'
+
+function LogsCsvPreview({ csvData }: { csvData?: CsvData }) {
+  return csvData ? (
+    <div className='animate-in fade-in duration-300 flex flex-col gap-2 max-h-80 overlo'>
+      <CsvPreviewTable csvData={csvData} />
+    </div>
+  ) : (
+    <div className='animate-in fade-in slide-in-from-top-5 duration-300 overflow-y-hidden'>
+      <TableSkeleton rows={8} cols={5} maxHeight={320} />
+    </div>
+  )
+}
+
+export function ExportLogsContent({
+  selectedLogsIds,
+  setCsvData,
+}: {
+  selectedLogsIds: number[]
+  setCsvData: (csvData?: CsvData) => void
+}) {
+  const fetcher = useFetcher(
+    ROUTES.api.documentLogs.generateCsv.detail({
+      documentLogIds: selectedLogsIds,
+    }).root,
+  )
+
+  const { data } = useSWR<CsvData>(['documentLogCsv', selectedLogsIds], fetcher)
+  useEffect(() => {
+    setCsvData(data)
+  }, [data, setCsvData])
```

Comment:
⚠️ Try to avoid `useEffect` if possible. In this case is an easy solution. Lift the `useSWR` collection up to the modal so you don't need to do it here and set the state callback